produce_request.go

```diff
@@ -50,8 +67,41 @@ func (r *ProduceRequest) encode(pe packetEncoder) error {
 			if err != nil {
 				return err
 			}
+			if metricRegistry != nil {
+				for _, messageBlock := range msgSet.Messages {
+					// Is this a fake "message" wrapping real messages?
+					if messageBlock.Msg.Set != nil {
+						topicRecordCount += int64(len(messageBlock.Msg.Set.Messages))
+					} else {
+						// A single uncompressed message
+						topicRecordCount++
+					}
+					// Better be safe than sorry when computing the compression ratio
+					if len(messageBlock.Msg.Value) != 0 {
+						compressionRatio := float64(messageBlock.Msg.payloadSize) /
```

Comment:
I followed the Java client for that metric, they named it `compression-rate-avg` (The average compression rate of record batches) and it is the inverse of a regular compression ratio indeed, computed as a float:
https://github.com/apache/kafka/blob/7115c66aefb11efa802e61a42bcc13fadf92598d/clients/src/main/java/org/apache/kafka/common/record/Compressor.java#L150

But I found it confusing the first time I tried to interpret that metric from the Java client to be honest so I can change it to expose 200 for representing a compression ratio of 2.0 instead of 50.

Renaming `payloadSize` to `compressedPayloadSize` make sense.